pkgs/development/python-modules/treeo/default.nix

```diff
@@ -4,41 +4,41 @@
 , jaxlib
 , lib
 , poetry-core
+, pytestCheckHook
 }:
 
 buildPythonPackage rec {
   pname = "treeo";
-  version = "0.4.0";
+  version = "0.0.9";
   format = "pyproject";
 
   src = fetchFromGitHub {
     owner = "cgarciae";
     repo = pname;
     rev = version;
-    sha256 = "176r1kgsdlylvdrxmhnzni81p8m9cfnsn4wwn6fnmsgam2qbp76j";
+    hash = "sha256-Yk4KNE5BncYoEHWZSEeSgoNb0TLMRbs8kSovUEKR2Ek=";
   };
 
-  postPatch = ''
-    substituteInPlace pyproject.toml \
-      --replace 'typing-extensions = "^3.10.0"' 'typing-extensions = "*"'
-  '';
-
   nativeBuildInputs = [
     poetry-core
   ];
 
   # jax is not declared in the dependencies, but is necessary.
   propagatedBuildInputs = [
     jax
+    jaxlib
```

Comment:
Thank you! This should not have been in `propagatedBuildInputs` to begin with per https://nixos.wiki/wiki/JAX#How_do_I_package_JAX_libraries.3F.

Reply:
Wait, isn't that the opposite of what we should be doing here? IIUC it was correct before the change (= `jaxlib` used only in `checkInputs` and not in `propagatedBuildInputs`).

@dotlambda please keep it the way it was before (= _not_ propagating `jaxlib` and explicitly passing it in `checkInputs`) as otherwise it forces the specific version of `jaxlib` onto the user, see the link above for details.

Reply:
oops! I was way too tired when I wrote that! yes jaxlib should absolutely _not_ be in `propagatedBuildInputs`. Thanks for catching @ndl!

Reply:
Normally we use upstreams requirements as a good recommendations but many python packages use the wrong requires or force linters we don't want. If that package is required for this package to work correct then it can be propagated even if upstream has done something different.

Reply:
We have an established policy of not propagating `jaxlib` in the JAX ecosystem. Sometimes that means deviating from upstreams dependency lists, and creating issues upstream as well.